app/controllers/replies_controller.rb

```diff
@@ -101,11 +101,16 @@ def create
     reply = Reply.new(params[:reply])
     reply.user = current_user
 
-    if reply.post.try(:first_unread_for, current_user) && !params[:unread_warned]
-      # There are unreads and the user has not been warned.
-      params[:unread_warned] = true
-      flash[:error] = "There are unread replies not shown here. (Click 'post' again if you wish to ignore this warning.)"
-      preview(reply) and return
+    if reply.post.present?
+      last_seen_reply_id = params[:last_seen]
+      @unseen_replies = reply.post.replies.order('id asc').paginate(page: 1, per_page: 5)
+      @unseen_replies = @unseen_replies.where('id > ?', last_seen_reply_id) if last_seen_reply_id.present?
+      most_recent_unseen_reply = @unseen_replies.last
+      if most_recent_unseen_reply.present?
+        flash[:error] = "There have been #{@unseen_replies.count} new replies since you last viewed this post."
+        @last_seen_id = most_recent_unseen_reply.id
```

Comment:
As `last_seen_id` increases when you resubmit the form but it doesn't actually mark the replies *read*, resubmitting multiple times (and accordingly reading the unread replies in 5-per-submission increments) will not impact whether you've 'read' those replies, as far as the site is concerned. This seems weird. (Maybe just mark the post read up to the new `@last_seen_id`, to fix this?)